collection/collection_size_proxy.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 Juan Pablo Tosso
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collection
+
+import (
+	"regexp"
+	"strconv"
+
+	"github.com/corazawaf/coraza/v3/types"
+	"github.com/corazawaf/coraza/v3/types/variables"
+)
+
+// SizeProxy are used to connect the size
```

Comment:
Just wondering, is there a reason for an entire separate type vs adding `Size` to `CollectionProxy`? It seems a bit mysterious to see stubbed implementations of `Find` methods

Reply:
I don't like it, but ARGS_COMBINED_SIZE cannot be a proxy. Otherwise, users would be able to do SecRule ARGS_COMBINED_SIZE:param_name

The reason for the dummy Find methods is to make them all compatible, Proxies, maps, and single values. Otherwise, we would have to do type assertion to retrieve parameters, and that is slower.

I'm open to suggestions.

Reply:
Thanks for the background, the current approach seems good enough then though may take another look later. Adding the combined size to body processor seems like it could be putting the feature in a somewhat unrelated location.